week-4/E-arrays-of-objects/exercise-2.js

```diff
@@ -7,50 +7,81 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
 3) Print in the console all the destination names more than 300 kms far away and reachable by train.
 */
 
-
 var destination1 = {
-    destinationName: "Edinburgh",
-    distanceKms: 80,
-    transportations: ["car", "bus", "train"]
+  destinationName: "Edinburgh",
+  distanceKms: 80,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination2 = {
-    destinationName: "London",
-    distanceKms: 650,
-    transportations: ["car", "bus", "train"]
+  destinationName: "London",
+  distanceKms: 650,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination3 = {
-    destinationName: "Paris",
-    distanceKms: 900,
-    transportations: ["train", "plane"]
+  destinationName: "Paris",
+  distanceKms: 900,
+  transportations: ["train", "plane"]
 };
 
 var destination4 = {
-    destinationName: "Dublin",
-    distanceKms: 350,
-    transportations: ["plane", "ferry"]
+  destinationName: "Dublin",
+  distanceKms: 350,
+  transportations: ["plane", "ferry"]
 };
 
-var travelDestinations = [destination1, destination2, destination3, destination4];
+var travelDestinations = [
+  destination1,
+  destination2,
+  destination3,
+  destination4
+];
 
 /*
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
 
+var destinationNamesWithin500Kms = travelDestinations
+  .filter(function(destination) {
+    return destination.distanceKms < 500;
+  })
+  .map(function(destination) {
+    return destination.destinationName;
+  }); // Complete here
 
-var destinationNamesWithin500Kms = // Complete here
-
-var destinationNameReachableByFerry = // Complete here
-
-var destinationNamesMoreThan300KmsAwayByTrain = // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
-
+var destinationNameReachableByFerry = travelDestinations
+  .filter(function(destination) {
+    return destination.transportations.includes("ferry");
+  })
+  .map(function(destination) {
+    return destination.destinationName;
+  }); // Complete here
 
+var destinationNamesMoreThan300KmsAwayByTrain = travelDestinations
+  .filter(function(destination) {
+    return (
+      destination.transportations.includes("train") &&
+      destination.distanceKms > 300
+    );
+  })
+  .map(function(destination) {
+    return destination.destinationName;
+  }); // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
```

Comment:
Just wanted to say well done with this one, lots of people had difficulty with this. You've got a good solution.